internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -743,11 +735,25 @@ func shouldUpdateIntegrationTestGitResolver(integrationTestScenario *v1beta2.Int
 func getGitResolverUpdateMap(snapshot *applicationapiv1alpha1.Snapshot) map[string]string {
 	annotations := snapshot.GetAnnotations()
 	return map[string]string{
-		tektonconsts.TektonResolverGitParamURL:      urlToGitUrl(annotations[gitops.PipelineAsCodeGitSourceURLAnnotation]), // should have .git in url for consistency and compatibility
+		tektonconsts.TektonResolverGitParamURL:      h.UrlToGitUrl(annotations[gitops.PipelineAsCodeGitSourceURLAnnotation]), // should have .git in url for consistency and compatibility
 		tektonconsts.TektonResolverGitParamRevision: annotations[gitops.PipelineAsCodeSHAAnnotation],
 	}
 }
 
+// shouldUpdateIntegrationTasksGitResolver checks if the integration test resolver should be updated based on the source repo
+func shouldUpdateIntegrationTasksGitResolver(integrationTestScenario *v1beta2.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot) bool {
+	// only "pull-requests" are applicable
+	if gitops.IsSnapshotCreatedByPACPushEvent(snapshot) {
+		return false
+	}
+
+	// only integrationTestScenarios with the pipelineRun resourceKind can update their task references
+	if integrationTestScenario.Spec.ResolverRef.ResourceKind != tektonconsts.ResourceKindPipelineRun {
+		return false
+	}
+	return true
+}
+
```

Comment:
Problem:
This function returns true for ALL PR snapshots with PipelineRun resource kind
It doesn't check if the integration test scenario's tasks actually come from the PR's repo
This causes unrelated integration tests to have their task references overwritten
Tasks from the integration test repo get replaced with non-existent tasks from the component's PR repo
Impact: Integration PipelineRuns try to reference tasks that don't exist in the PR's repository, causing resolution failures